### PR TITLE
Implement PhasedXZGate._unitary_

### LINKED ISSUE #3965: PhasedXZGate has_channel evaluates to True but channel returns NotImplemented
**Description of the issue**
See code below
**How to reproduce the issue**

```python
import cirq
 
op = cirq.PhasedXZGate(axis_phase_exponent=-0.5, x_exponent=-0.5,
                       z_exponent=1).on(cirq.GridQubit(3, 3))
assert cirq.has_channel(op)
try:
    cirq.channel(op)
except TypeError as error:
    print(f'Got error {error}')
```

<details>
Print statement reads:

Got error object of type '<class 'cirq.ops.gate_operation.GateOperation'>' does have a _channel_, _mixture_ or _unitary_ method, but it returned NotImplemented.

</details>

**Cirq version**
0.9.0
0.10.0.dev

### LINKED ISSUE #4434: cirq.kraus seems to fail on PhasedXZGate
**Description of the issue**
I get a TypeError when using cirq.kraus on this gate.
**How to reproduce the issue**

```python
import cirq

op = cirq.PhasedXZGate(axis_phase_exponent=-0.5,
                           x_exponent=-0.5,
                           z_exponent=1).on(cirq.GridQubit(3, 3))
cirq.kraus(op)
```
Produces:
```
Traceback (most recent call last):
  File "/usr/local/google/home/dkafri/.config/JetBrains/PyCharmCE2021.2/scratches/scratch_62.py", line 6, in <module>
    cirq.kraus(op)
  File "/usr/local/google/home/dkafri/.virtualenvs/pyle/lib/python3.8/site-packages/cirq/protocols/kraus.py", line 189, in kraus
    raise TypeError(
TypeError: object of type '<class 'cirq.ops.gate_operation.GateOperation'>' does have a _kraus_, _mixture_ or _unitary_ method, but it returned NotImplemented.
```

**Cirq version**
I get this error both for `0.12.0.dev20210630205305` and `0.12.0`

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #3965. Fixes #4434.

### CODE REVIEW

File: cirq-core/cirq/ops/phased_x_z_gate_test.py [hunk 1]
```diff
@@ -199,6 +199,22 @@ def test_from_matrix_close_unitary(unitary: np.ndarray):
...
+        cirq.testing.random_unitary(2),
+        cirq.testing.random_unitary(2),
+        np.array([[0, 1], [1j, 0]]),
+    ],
+)
+def test_from_matrix_close_kraus(unitary: np.ndarray):
+    gate = cirq.PhasedXZGate.from_matrix(unitary)
+    kraus = cirq.kraus(gate)
+    assert len(kraus) == 1
+    cirq.testing.assert_allclose_up_to_global_phase(kraus[0], unitary, atol=1e-8)
```
Comment (MichaelBroughton) [inline cirq-core/cirq/ops/phased_x_z_gate_test.py:215]: Question: is this tol too high for complex64. I would've expected only around 6-7 decimal digits of reliability.

File: cirq-core/cirq/ops/phased_x_z_gate_test.py [hunk 2]
```diff
@@ -199,6 +199,22 @@ def test_from_matrix_close_unitary(unitary: np.ndarray):
...
+@pytest.mark.parametrize(
+    'unitary',
+    [
+        cirq.testing.random_unitary(2),
+        cirq.testing.random_unitary(2),
+        cirq.testing.random_unitary(2),
+        np.array([[0, 1], [1j, 0]]),
+    ],
+)
+def test_from_matrix_close_kraus(unitary: np.ndarray):
```
Comment (MichaelBroughton) [inline cirq-core/cirq/ops/phased_x_z_gate_test.py:211]: Can we add another test that hits `cirq.unitary` as well ?

**Review by MichaelBroughton — APPROVED** [review]
LGTM with some questions.

File: cirq-core/cirq/ops/phased_x_z_gate_test.py (hunk 2 above)
Comment (maffoo) [reply cirq-core/cirq/ops/phased_x_z_gate_test.py:211]: There is already a test of `cirq.unitary`: https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/ops/phased_x_z_gate_test.py#L196. This worked by decomposition even without a `_unitary_` implementation, but other protocols like `cirq.kraus` didn't.

File: cirq-core/cirq/ops/phased_x_z_gate_test.py (hunk 1 above)
Comment (maffoo) [reply cirq-core/cirq/ops/phased_x_z_gate_test.py:215]: This was copy-pasted from other `allclose` checks in the same module. I'm inclined to keep it for consistency, but we could certainly relax it throughout the module if desired.

**Pull request merged**